README.md

```diff
@@ -10,29 +10,49 @@ This repo contains the AWS ParallelCluster Chef cookbook used in AWS ParallelClu
-You can find more information about them in the `cookbooks/third-party/THIRD-PARTY-LICENSES.txt` file.
+the other sub-cookbooks and third party cookbooks.
+
+The main cookbook does not contain any recipe, attribute or library. They are distributed in the functional cookbooks under the `cookbooks` folder
+defined as follows:
+- `aws-parallelcluster-entrypoints` is the cookbook to define the external interface, contains recipes called by AMI builder, cluster setup, cluster update, etc. 
+  The recipes in this cookbook are called directly by the CLI or CI/CD. It orchestrates invocation of recipes/resources from other cookbooks;
+- `aws-parallelcluster-platform` OS packages and system configuration (directories, users, services, drivers);
+- `aws-parallelcluster-environment` AWS services configuration and usage, such as shared file systems, directory service and network interfaces;
+- `aws-parallelcluster-computefleet` scaling logic, compute fleet scripts and daemons;
+- `aws-parallelcluster-awsbatch` files required to support AWS Batch as scheduler;
+- `aws-parallelcluster-slurm` files required to support Slurm as a scheduler and its dependencies (Munge, MySQL for accounting, etc);
+
+Finally, some common code, such as source/script directories, usernames, package installer etc., are located in a cookbook 
+which every other cookbook depend on, that is `aws-parallelcluster-shared`.
+Each cookbook hosts recipes and resources, attributes, functions, files and templates belonging to its functional area.
+
+Every cookbook contains ChefSpec and Kitchen tests for its code.
+However, the code in a cookbook might require that some other code from a different cookbook not listed among dependencies,
+to be executed as a prerequisite (test setup phase). For this reason `aws-parallelcluster-tests` cookbook must depend on every other cookbook.
+
+The `test` folder contains Python unit tests files and Kitchen [environment](https://docs.chef.io/environments/) files.
 
 The `kitchen` folder contains utility files used when running Inspec tests locally.
 
-The `test` folder contains Python unit tests files and Kitchen [environment](https://docs.chef.io/environments/) files.
+The `cookbooks/third-party` folder contains cookbooks from marketplace. They must be regularly updated and should not be modified by hand.
+You can find more information about them in the `cookbooks/third-party/THIRD-PARTY-LICENSES.txt` file.
 
 # Development
 
-## About kitchen tests
+## About ChefSpec Tests
+
+[ChefSpec](https://github.com/chefspec/chefspec) is a unit testing framework for testing Chef cookbooks. 
+It is very fast, and we use it to verify recipes with multiple branches (e.g. HeadNode vs ComputeNode) work as expected.
+They don't need virtual machines or cloud servers. They can be executed locally by executing:
+
+```
+cd cookbooks/aws-parallelcluster-platform
+chef exec rspec ./spec/unit/recipes/sudo_config_spec.rb 
+```
+
+They are automatically executed as GitHub actions.
```

Comment:
Did you check all of them have been added to GH actions?

Reply:
We were missing awsbatch, adding it.

Reply:
done here: https://github.com/aws/aws-parallelcluster-cookbook/pull/2318